Download and install the TensorFlow 2.0 Beta package. Import TensorFlow into your program:

In [1]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
try:
  # %tensorflow_version only exists in Colab.
  %tensorflow_version 2.x
except Exception:
  pass

import tensorflow as tf

Load and prepare the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). Convert the samples from integers to floating-point numbers:

In [2]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

Build the `tf.keras.Sequential` model by stacking layers. Choose an optimizer and loss function for training:

In [3]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate the model:

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

Epoch 1/5
1875/1875 [==============================] - 17s 8ms/step - loss: 0.2890 - accuracy: 0.9158
Epoch 2/5
1875/1875 [==============================] - 18s 10ms/step - loss: 0.1393 - accuracy: 0.9582
Epoch 3/5
1875/1875 [==============================] - 24s 13ms/step - loss: 0.1042 - accuracy: 0.9682
Epoch 4/5
1875/1875 [==============================] - 18s 10ms/step - loss: 0.0847 - accuracy: 0.9741
Epoch 5/5
313/313 [==============================] - 3s 8ms/step - loss: 0.0751 - accuracy: 0.9776


[0.0751296654343605, 0.9775999784469604]

The image classifier is now trained to ~98% accuracy on this dataset. To learn more, read the [TensorFlow tutorials](https://www.tensorflow.org/beta/tutorials/).